engines/sci/sound/decoders/sol.cpp

```diff
@@ -112,16 +115,124 @@ static void deDPCM8Nibble(int16 *out, uint8 &sample, uint8 delta) {
 	*out = ((lastSample + sample) << 7) ^ 0x8000;
 }
 
+/**
+ * Decompresses one half of an 8-bit DPCM compressed audio
+ * byte. Attempts to repair overflows on the fly.
+ */
+static void deDPCM8NibbleWithRepair(int16 *const out, uint8 &sample, const uint8 delta,
+									uint8 &repairState, uint8 &preRepairSample) {
+	const uint8 lastSample = sample;
+
+	// In Gabriel Knight: Sins of the Fathers CD, the DPCM8-encoded speech contains overflows.
+	// Overflows wrap from positive to negative, or negative to positive. This continues
+	// until the wave settles at a new (wrapped) zero DC offset.
+
+	// We can't look ahead to see where the wave "reconnects" to valid data, so we
+	// decay the wave on an artificial slope until it does. This seems to take on average
+	// around 9-10 samples. The slope value below was chosen through analysing spectrographs
+	// of the decoded/repaired wave data to find the slope which removed the pops most
+	// cleanly across a test selection of game speech.
+
+#define REPAIR_SLOPE 12
+//#define SHOW_REPAIR_EVENTS
+
+	switch (repairState) {
+	default:
+		assert(false); // Invalid state.
+		// Fall through...
+	case 0: {
+		const int16 newSampleOverflow = (int16)sample + tableDPCM8[delta & 15];
+
+		if (newSampleOverflow > 255) {
+			// Positive overflow has occurred; begin artificial negative slope.
+			repairState = 1;
+			sample = lastSample - REPAIR_SLOPE;
+			// We also begin tracking the un-repaired waveform, so we can tell when to stop.
+			preRepairSample = (uint8)newSampleOverflow;
+
+#ifdef SHOW_REPAIR_EVENTS
+			warning("DPCM8 OVERFLOW (+)");
+#endif
+
+		} else if (newSampleOverflow < 0) {
+			// Negative overflow has occurred; begin artificial positive slope.
+			repairState = 2;
+			sample = lastSample + REPAIR_SLOPE;
+			// We also begin tracking the un-repaired waveform, so we can tell when to stop.
+			preRepairSample = (uint8)newSampleOverflow;
+
+#ifdef SHOW_REPAIR_EVENTS
+			warning("DPCM8 OVERFLOW (-)");
+#endif
+
+		} else {
+			sample = (uint8)newSampleOverflow;
+		}
+	}	break;
+	case 1: {
+		// Check for a slope wrap. This circumstance should never happen in reality;
```

Comment:
Same assert story from above. If it's impossible, `error`, if you have doubt then I'd rather a sound didn't play for a user with a logged warning than crash their game.